test/e2e/fixtures.go

```diff
@@ -192,6 +192,22 @@ func exportLogs(tb testing.TB, data *TestData) {
 		return nil
 	})
 
+	// dump the logs for monitoring Pods to disk.
+	data.forAllMonitoringPods(func(nodeName, podName string) error {
+		w := getPodWriter(nodeName, podName, "logs")
+		if w == nil {
+			return nil
+		}
+		defer w.Close()
+		cmd := fmt.Sprintf("kubectl -n %s logs --all-containers %s", monitoringNamespace, podName)
+		stdout := runKubectl(cmd)
+		if stdout == "" {
+			return nil
+		}
+		w.WriteString(stdout)
+		return nil
```

Comment:
this is exactly the same code as above I think, so we could define a local function to do that:
```go
writePodLogs := func(nodeName, podName string) error {
  // ...
}
data.forAllAntreaPods(writePodLogs)
data.forAllMonitoringPods(writePodLogs)
```

Reply:
Indeed, it took some refactoring though.